apps/dc_tools/odc/apps/dc_tools/_stac.py

```diff
@@ -98,30 +99,27 @@ def _stac_product_lookup(
     if constellation is not None:
         constellation = constellation.lower().replace(" ", "-")
 
-    # Let's check for Sentinel-2, which is a special case
-    # The below section handles the Element 84 and Microsft PC STAC documents
```

Comment:
Do these comments need to be preserved? Seems to be the justification for the hack. Ideally there'd be test cases for these special cases...

Reply:
I don't think it does what it claims. There is no justification to hardcode the product name. After the change, that segment was limited to DEAfrica sentinel 2 product. Hence it doesn't have any general meaning to cater special format of stac documents. Had it been the point, it should have been coded in `stac_transform` to handle the format rather a single product.